src/Components/QuickGrid/Microsoft.AspNetCore.Components.QuickGrid/test/GridRaceConditionTest.cs

```diff
@@ -0,0 +1,144 @@
+    {
+        var testComponent = new TestComponent();
+
+        var componentId = _renderer.AssignRootComponentId(testComponent);
+        _renderer.RenderRootComponent(componentId);
+        await Task.Delay(10);
+        _renderer.RenderRootComponent(componentId);
+        _tcs.SetResult();
+    }
+}
+
+internal class TestComponent : ComponentBase
+{
+    private bool _firstRender = true;
+    private PaginationState _pagination = new() { ItemsPerPage = 2 };
+
+    internal class Person
+    {
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
+        public int Age { get; set; }
+    }
+
+    private IQueryable<Person> _people = new List<Person>
+    {
+        new Person { Id = 1, Name = "John Doe", Age = 30 },
+        new Person { Id = 2, Name = "Jane Smith", Age = 25 },
+        new Person { Id = 3, Name = "Alice Johnson", Age = 22 }
+    }.AsQueryable();
+
+    protected override void BuildRenderTree(RenderTreeBuilder builder)
+    {
+        if (_firstRender)
+        {
+            //Render the QuickGrid
+            builder.OpenComponent<QuickGrid<Person>>(0);
+            builder.AddAttribute(1, "Items", _people);
+            builder.AddAttribute(2, "Pagination", _pagination);
+            builder.AddAttribute(3, nameof(QuickGrid<Person>.ChildContent),
+                (RenderFragment)(builder => BuildColumnsRenderFragment(builder)));
+            builder.CloseComponent();
+
+            builder.OpenComponent<Paginator>(4);
+            builder.AddAttribute(5, "State", _pagination);
+            builder.CloseComponent();
+            _firstRender = false;
+        }
+    }
+
+    protected void BuildColumnsRenderFragment(RenderTreeBuilder builder)
+    {
+        //Render the PropertyColumn for Id
+        builder.OpenComponent<PropertyColumn<Person, int>>(0);
+        builder.AddAttribute(1, nameof(PropertyColumn<Person, int>.Property),
+            (System.Linq.Expressions.Expression<Func<Person, int>>)(p => p.Id));
+        builder.AddAttribute(2, nameof(PropertyColumn<Person, int>.Sortable), true);
+        builder.CloseComponent();
+
+        //Render the PropertyColumn for Name
+        builder.OpenComponent<PropertyColumn<Person, string>>(3);
+        builder.AddAttribute(4, nameof(PropertyColumn<Person, string>.Property),
+            (System.Linq.Expressions.Expression<Func<Person, string>>)(p => p.Name));
+        builder.AddAttribute(5, nameof(PropertyColumn<Person, string>.Sortable), true);
+        builder.CloseComponent();
+
+        //Render the PropertyColumn for Age
+        builder.OpenComponent<PropertyColumn<Person, int>>(6);
+        builder.AddAttribute(7, nameof(PropertyColumn<Person, int>.Property),
+            (System.Linq.Expressions.Expression<Func<Person, int>>)(p => p.Age));
+        builder.AddAttribute(8, nameof(PropertyColumn<Person, int>.Sortable), true);
+        builder.CloseComponent();
+    }
+
+    public void TriggerRender()
```

Comment:
When is `TriggerRender` called? Let's remove all the changes that are not directly connected to testing the bug.